PWMoutputMATLABtutorial

MessiahJa opened the pull request:
This a contribution. While I will be completing the tutorial based on the guidelines (such as adding the "Ciruit" section), I have a few questions:

- Should I convert and upload all images as 1920x1080 png? Or will you be providing the images with a certain theme?
- The introduction for MATLAB was taken from the [nanoMatlabWiFiLED](https://github.com/arduino/docs-content/blob/main/content/hardware/03.nano/carriers/nano-motor-carrier/tutorials/nano-matlab-wifi-led/nanoMatlabWiFiLED.md) tutorial. Should it be changed and merged into a part of the main section of the tutorial?

## Contribution Guidelines
- [x] I confirm that I have read the [contribution guidelines](https://github.com/arduino/docs-content/tree/main/contribution-templates) and comply with them.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/arduino/docs-content?pullRequest=222) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/arduino/docs-content?pullRequest=222) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/arduino/docs-content?pullRequest=222) it.</sub>

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+The goals of this project are:
+
+- Control the Arduino Uno LED with MATLAB.
```
```suggestion
- Control the UNO R3 LED with MATLAB.
```

jhansson-ard replied:
After the first time mention of the full name of the product, it is ok to used the product name without the 'Arduino®'

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+The goals of this project are:
+
+- Control the Arduino Uno LED with MATLAB.
+- Control the Arduino Uno PWM output with MATLAB.
```
```suggestion
- Control the UNO R3 PWM output with MATLAB.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+### Hardware & Software Needed
+
+- [Arduino Uno Rev3](https://store.arduino.cc/products/arduino-uno-rev3)
```
```suggestion
- [UNO R3](https://store.arduino.cc/products/arduino-uno-rev3)
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+---
+title: 'Controlling PWM output with a potentiometer on the Arduino Uno via MATLAB'
```
```suggestion
title: 'Controlling PWM output with a potentiometer on the Arduino® UNO R3 via MATLAB'
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+- [MATLAB® Support Package for Arduino® Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware)
+
+
+***The Arduino Uno, potentiometer, USB 2.0 Cable Type A/B and jumper wires are included as part of the [Arduino Starter Kit](https://store.arduino.cc/collections/kits/products/arduino-starter-kit-multi-language)***
```
```suggestion
***The UNO R3, potentiometer, USB 2.0 Cable Type A/B and jumper wires are included as part of the [Arduino Starter Kit](https://store.arduino.cc/collections/kits/products/arduino-starter-kit-multi-language)***
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+ 
+
+### LED on the Arduino Uno R3
+The internal LED in the Arduino Uno is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.
```
```suggestion
The internal LED in the UNO R3 is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+![Selection of Connection Type](assets/connectiontype.png)
+
+
+**4.** Choose the "Uno" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.
```
```suggestion
**4.** Choose the "UNO" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+![arduino Setup](assets/arduinosetup.png)
+
+**3.** A Graphical User Interface (GUI) appears, that will help you set up the MATLAB-Arduino connection. To establish the connection over the USB cable, make sure that the USB radio box is selected and then click on Next.
```
```suggestion
**3.** A Graphical User Interface (GUI) appears, that will help you set up the MATLAB®-Arduino connection. To establish the connection over the USB cable, make sure that the USB radio box is selected and then click on Next.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+</p>
+
+
+**5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.
```
```suggestion
**5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB® connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+<img src="https://i.imgur.com/QEXZ27s.png"/>
+</p>
+
+**6.** The Arduino Uno is now configured to interact with MATLAB. Now to create the object in MATLAB, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
```
```suggestion
**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+![](assets/arduinoproperties.png)
+
+**7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
```
```suggestion
**7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+## Set up the circuit and code
+
+**1.** On one breadboard we have the white jumper wire sending 5 volts to the potentiometer. The orange cable is connected to GND(Ground) and the gray wire is our input to A0. A0 will read the variable voltage coming in from pin13 after being adjusted through the potentiometer.
+On the right breadboard we have a simple LED circuit comprised of a red output wire connected to pin11, a resistor and the LED finally connected to GND via the green wire
```
```suggestion
On the right breadboard we have a simple LED circuit comprised of a red output wire connected to pin11, a resistor and the LED finally connected to GND via the green wire.
```

jhansson-ard replied:
Missing punctuation

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+***Make sure to connect the longer leg of the LED to the resistor and the shorter to GND.***
+
+
+**2.** Set your output voltage to 1 in the previously mentioned way, using `writeDigitalPin(a, 'D13', 1);` Now we'll check to see if A0 is actually reading an adjustable voltage. To see if the input changes as we turn the potentiometer use `readVoltage(a, A0);` in a loop.
```
```suggestion
**2.** Set your output voltage to 1 in the previously mentioned way, using `writeDigitalPin(a, 'D13', 1);` Now we'll check to see if A0 is actually reading an adjustable voltage. To see if the input changes as we turn the potentiometer use `readVoltage(a, A0);` in a loop.

```

jhansson-ard replied:
we should always add a space before each code snippet

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+MATLAB should be displaying different amount for "Volts as your turn the potentiometer.
+</p>
+
+**3.** All that is left is to send the amount recieved from A0 to pin 11. For this we need a different command: `writePWMVolatge()`.
```
```suggestion
**3.** All that is left is to send the amount recieved from A0 to pin 11. For this we need a different command: `writePWMVolatge()`.

```

jhansson-ard replied:
we should always add a space before each code snippet

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+## Further Ideas
+
+- You can also use the command line arguments to upload the MATLAB server to the Nano 33 IoT via the `arduino()` command. See more information on using this function in the [MathWorks documentation](https://www.mathworks.com/help/supportpkg/arduinoio/ref/arduino.html).
+
```
I suggest to remove the space between the bullet points to make the list coherent

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+## Introduction
+
+In this tutorial, we will use MATLAB to turn on an external LED, and then to adjust the output using MATLAB® Support Package for Arduino® Hardware.
```
```suggestion
In this tutorial, we will use MATLAB® to turn on an external LED, and then to adjust the output using MATLAB® Support Package for Arduino® Hardware.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+</p>
+
+<p align="center">
+MATLAB should be displaying different amount for "Volts as your turn the potentiometer.
```
```suggestion
MATLAB® should be displaying different amount for "Volts as your turn the potentiometer.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+- You can also use the command line arguments to upload the MATLAB server to the Nano 33 IoT via the `arduino()` command. See more information on using this function in the [MathWorks documentation](https://www.mathworks.com/help/supportpkg/arduinoio/ref/arduino.html).
+
+- Try writing a MATLAB code that slowly increases and then decreases the blink speed instead of adjusting the brightness.
```
```suggestion
- Try writing a MATLAB® code that slowly increases and then decreases the blink speed instead of adjusting the brightness.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+
+
+### Why MATLAB?
+MATLAB® is an educational and industrial programming platform used to analyse data, perform simulations and carry out model based designs. Through an interactive communication with an Arduino board, you can expand MATLAB's capabilities, while also gaining access to a wide range of math, engineering and plotting functions. Check out the capabilities of MATLAB® over on the [MathWorks website](https ...
```
Updated the MATLAB® and MathWorks® trademark

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+The internal LED in the Arduino Uno is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.
+
+<p align="center">
+<img src="http://electronicswithyou.com/wp-content/uploads/2019/05/Arduino-Uno-LED.jpg"/>
```
Please use the markdown formatting instead, for example:

![image description](link.com/url/to/image.jpg)

MessiahJa replied:
Will do.

jhansson-ard reviewed (commented):
@MessiahJa thank you for your contribution! I made a small QA and was mainly focusing on the trademarks and formatting of your tutorial. 

Please take a look and let me know what you think. If you agree (which you should at least with the trademarks), please use the "commit suggestion" button to implement my suggestions!

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,147 @@
+title: 'Controlling PWM output with a potentiometer on the Arduino Uno via MATLAB'
+difficulty: intermediate
+compatible-products: [Arduino Uno Rev 3, Potentiometer]
+description: 'Learn how to dim an LED connceted to the Arduino Uno with MATLAB.'
```
```suggestion
description: 'Learn how to dim an LED connceted to the UNO R3 with MATLAB.'
```

jhansson-ard commented:
Then, to answer your questions.

1. Yes, please do make sure all your images are in the format 1920x1080. If you are using external images, try using images or format them in some way that they also follow the guidelines. You images look fine, and we  don't need to remake them!
2. If the introduction still makes sense, you can leave it. However if it is somehow different from the tutorial you copied, please add a sentence or two about what your tutorial is about.

again, thank you for your contribution! ⭐

MessiahJa reviewed (commented):
I'll retake the screenshots in 1080.

per1234 commented:
Hi @MessiahJa. Please sync your fork by following this tutorial:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Your fork is currently very outdated compared to Arduino's repository and that is causing the automated "Content Linter" check to fail. It should be as simple to do as clicking the button shown in the tutorial I linked above.

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+---
+title: 'Controlling PWM output with a potentiometer on the Arduino® UNO R3 via MATLAB®'
```
```suggestion
title: 'Controlling PWM Output with a Potentiometer via MATLAB®'
```

Fix content linter failures:

- Make title <=60 characters
  - My thinking is this tutorial is already categorized under the Arduino Uno board section, so that information is available to the reader in other ways
- Use [title case](https://en.wikipedia.org/wiki/Title_case)

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+title: 'Controlling PWM output with a potentiometer on the Arduino® UNO R3 via MATLAB®'
+difficulty: intermediate
+compatible-products: [Arduino Uno Rev 3, Potentiometer]
+description: 'Learn how to dim an LED connceted to the UNO R3 with MATLAB®.'
```
```suggestion
description: 'Learn how to dim an LED connected to the UNO R3 with MATLAB®.'
```
Fix typo

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+
+**9.** See what happens when you change the pause duration. Try replacing the while with an `if` statement to make it blink a certain number of times.
+
+## Set up the circuit and code
```
```suggestion
## Set up the Circuit and Code
```
Use title case

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+### LED on the Arduino Uno R3
+The internal LED in the UNO R3 is connected to Digital Pin 13. By pulling this pin HIGH, we can turn the LED on and by pulling it LOW we can turn the LED off. But to output a variable voltage we have to use pins such as pin ~11 or ~10.
+
+![Pin13](assets/Arduino-Uno-LED.jpg)
```
```suggestion
![Pin 13](assets/Arduino-Uno-LED.jpg)
```
Fix the "Image doesn't have a (proper) description" content linter error

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+
+**4.** Choose the "UNO" from the dropdown menu, as well as the relevant COM port, and click on the blue `Program` button to upload the Arduino Server to the board. When you have done so, click on next.
+
+![Programming-the-arduino](assets/program-Arduino.png)
```
```suggestion
![Programming the Arduino](assets/program-Arduino.png)
```
Fix the "Image doesn't have a (proper) description" content linter error

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+**5.** In this step you can review the connection type, Port, board and loaded libraries. You can also click on Test connection to evaluate the Arduino-MATLAB® connection. Next, you should see a Green check-mark signalling the successful connection, as shown in the image below.
+
+
+![Program-successful](assets/programming-done.png)
```
```suggestion
![Programming successful](assets/programming-done.png)
```
Fix the "Image doesn't have a (proper) description" content linter error

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,154 @@
+
+**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
+
+![](assets/arduinoproperties.png)
```
```suggestion
![Arduino properties](assets/arduinoproperties.png)
```
Fix the "Image doesn't have a (proper) description" content linter error

MessiahJa commented:
@per1234 I was making new changes at the same time as your corrections so I could not commit your suggestions since they became outdated. I corrected the spelling errors, shortened the title and added "proper" descriptions to all images. Thank you.

per1234 commented:
You are welcome. I find GitHub's "suggestions" feature useful for unambiguous communication during PR reviews, but the PR author is always welcome to implement my suggestions however they like (if at all).

Please check the results of the latest CI run. There is good progress toward a ✔️, but still a few problems that will need to be resolved:

https://github.com/arduino/docs-content/runs/6434948735?check_suite_focus=true#step:3:9

```text
./content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:97: curcuit ==> circuit
./content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:142: porcessed ==> processed
./content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:142: ouput ==> output
```

https://github.com/arduino/docs-content/runs/6434948755?check_suite_focus=true#step:3:43

```text
❌ ERROR: 'Connecting the board to MATLAB' is not title case. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:[45](https://github.com/arduino/docs-content/runs/6434948755?check_suite_focus=true#step:3:45):1
❌ ERROR: 'Programming the board' is not title case. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:101:1
❌ ERROR: Asset 'Arduino-Connect.png' is not used. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:1:1
❌ ERROR: Asset 'Arduino-Uno-LED.jpg' is not used. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:1:1
❌ ERROR: Asset 'arduinoproperties.png' is not used. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:1:1
❌ ERROR: Asset 'curcuit.jpg' is not used. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:1:1
❌ ERROR: Asset 'program-Arduino.png' is not used. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:1:1
❌ ERROR: Asset 'programming-done.png' is not used. Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:1:1
❌ ERROR: Image doesn't have a (proper) description: assets/curcuit.png Location: ../../content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:97:1
```

It will also be required for you to sign the contributor license agreement offered by the bot here:

https://github.com/arduino/docs-content/pull/222#issuecomment-1123793350

MessiahJa commented:
@per1234 Is there any tutorial that could show me how I could use CI? I saw that the results do not update and it has to run again but I've not found any tutorials on it.

per1234 commented:
>Is there any tutorial that could show me how I could use CI

There is general information about it here:

https://docs.github.com/en/actions

However, that is only about the "GitHub Actions" framework the continuous integration system of this repository runs on. There isn't anything there specific to Arduino's own checks.

The "Spell Check" check that runs is from this excellent tool named codespell:

https://github.com/codespell-project/codespell

If you install that on your computer, you could run codespell on the projects yourself using a command like this:

```text
codespell -I "scripts/resources/spell-check-ignore-list.txt" --skip="*.svg,*.dxf" ./content/
```

The "Content Linter" check is a custom tool created by Arduino and hosted here:

https://github.com/arduino/content-lint

>I saw that the results do not update and it has to run again but I've not found any tutorials on it.

Unfortunately, there is nothing you can do about that. Arduino has configured the repository so that a maintainer must approve the CI runs for PRs from new contributors. That means there is an unfortunate delay between you pushing a new commit to the pull request and when one of the maintainers notice that an approval is required.

If you submit additional pull requests here after this one has been merged, the CI system will run the checks automatically.

marqdevx commented:
Should we close this PR? #193 
@MessiahJa

MessiahJa commented:
@marqdevx If nothing is left to fix then yes! :D

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+## Conclusion
+
+You can now control your output using variable inputs to control your devices and make use of the powerful features for scientific computing and developing engineering applications! Different input modulators similar to the potentiometer processed by the various toolboxes offered my MATLAB can lead to infinite types of ouput beyond that of the brightness of an LED.
```
```suggestion
You can now control your output using variable inputs to control your devices and make use of the powerful features for scientific computing and developing engineering applications! Different input modulators similar to the potentiometer processed by the various toolboxes offered my MATLAB can lead to infinite types of output beyond that of the brightness of an LED.
```

jhansson-ard commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+On one breadboard there is a white jumper wire sending 5 volts to the potentiometer. The orange cable is connected to GND(Ground) and the gray wire is our input to A0. A0 will read the variable voltage coming in from pin13 after being adjusted through the potentiometer.
+On the right breadboard we have a simple LED circuit comprised of a red output wire from pin11, a resistor and the LED finally connected to GND via the green wire.
+
+![Circuit](assets/circuit.png)
```
```suggestion
![Full circuit of the setup.](assets/circuit.png)
```

jhansson-ard replied:
@MessiahJa found a last image alternative text here, if you agree with the description - please accept the change 🙏🏼

jhansson-ard commented:
@per1234 if you are happy with the tutorial now, could you please give an approving review? We are blocked from merging atm ⭐

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+![Voltage display](assets/Volts.PNG)
+
+**2.** Set the value of `Volts` as the voltage to be sent through pin 11 using the command: `writePWMVolatge()`. This command is different from `writeDigitalPin()` as it's voltage is adjustable between 1 and 5 volts. Whereas the latter is either ON(1), being **5V** or OFF(0), **0V**.
```
```suggestion
**2.** Set the value of `Volts` as the voltage to be sent through pin 11 using the command: `writePWMVoltage()`. This command is different from `writeDigitalPin()` as its voltage is adjustable between 1 and 5 volts. Whereas the latter is either ON(1), being **5V** or OFF(0), **0V**.
```

https://www.mathworks.com/help/supportpkg/arduinoio/ref/writepwmvoltage.html

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+writeDigitalPin(a, 'D13', 1)
+while(1)
+  Volts = readVoltage(a, 'A0');
+  writePWMVolatge(a, 'D11', Volts)
```
```suggestion
  writePWMVoltage(a, 'D11', Volts)
```

per1234 commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
````diff
@@ -0,0 +1,148 @@
+
+**8.** Now, you can use this command to continuously blink an LED as part of a `while` loop. Try entering the script below.
+
+```arduino
````
````suggestion
```
````
This code is not written in the Arduino language, so it should not be syntax highlighted according to the rules of that language.

I see that the Linguist library used by GitHub does have a `MATLAB` identifier. I'm not sure which one, if any is used by the Arduino website system though.

https://github.com/github/linguist/blob/v7.20.0/lib/linguist/languages.yml#L3553

per1234 reviewed (approved):
I don't have any experience with MATLAT and don't possess a MATLAB license, so I am not able to fully evaluate the tutorial, but the parts I am able to evaluate look fine.

Thanks @MessiahJa!

MessiahJa commented:
@per1234 Thanks for your evaluation!

marqdevx commented:
@jhansson-ard Should we remake the pictures with an schematic view?

marqdevx commented:
Also I've tried to render it locally and everything seems good, no issues on the formatting side. ✔️

marqdevx commented:
@MessiahJa We are making some schematics pictures to replace the photos you took.

Also next week I will make some small changes, we will wait for your approval before merging it. 😄

MessiahJa commented:
Hello @marqdevx !
Is there anything I can do to help out with the changes and schematics? I have worked a lot on this and would love to see it included in the Arduino Docs. I'm working on a new one and want to have this merged before pulling the next tutorial.

marqdevx commented:
Sorry, I couldn't work on it yet.
I want just to make some small changes to fit our style guide, any major change.
I will jump on it ASAP

MessiahJa commented:
Its been more than a month since I last followed up. I put a lot of time and effort into creating this and was hoping to make a bunch more back to back but theres no response from your team. @marqdevx

marqdevx commented:
@MessiahJa Sorry for this time having your PR on hold, we will try to improve our timings for the community PRs.

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+- [UNO R3](https://store.arduino.cc/products/arduino-uno-rev3)
+- [USB 2.0 Cable Type A/B](https://store.arduino.cc/products/usb-2-0-cable-type-a-b)
+- Any potentiometer connectable to a breadboard
```
```suggestion
- Potentiometer trough-hole
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+- [MATLAB® Support Package for Arduino® Hardware](https://www.mathworks.com/matlabcentral/fileexchange/47522-matlab-support-package-for-arduino-hardware)
+
+
+***The UNO R3, potentiometer, USB 2.0 Cable Type A/B and jumper wires are included as part of the [Arduino Starter Kit](https://store.arduino.cc/collections/kits/products/arduino-starter-kit-multi-language)***
```
Not needed
```suggestion
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+![Programming successful](assets/SetupSuccess1080.PNG)
+
+
+**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a=arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
```
```suggestion
**6.** The UNO R3 is now configured to interact with MATLAB®. Now to create the object in MATLAB®, we run the command `a = arduino()`. The properties of the object, including the COM port, are displayed as shown in the example below.
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+![Arduino properties](assets/arduinoproperties1080.png)
+
+**7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
```
```suggestion
**7.** Enter the command `writeDigitalPin(a, 'D13', 1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 **D13** is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+**7.** Enter the command `writeDigitalPin(a,'D13',1);`. This command is similar to the `digitalWrite(D13, HIGH)` by the Arduino programming language. Which means that digital pin 13 (D13) is connected to the built-in LED, `a` is the Arduino object we created, and 1 represents a HIGH or ON state. Try turning the LED ON and OFF several times by changing 1 to 0 and vice-versa.
+
+
+***Unlike in the Arduino IDE, here you cannot address the built-in LED using the `LED_BUILTIN` function.***
```
```suggestion
***Unlike in the Arduino IDE, here you cannot address the built-in LED using the `LED_BUILTIN` macro.***
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
````diff
@@ -0,0 +1,148 @@
+end
+```
+
+**9.** See what happens when you change the pause duration. Try replacing the while with an `if` statement to make it blink a certain number of times.
````
Confusing for the end user, reword, change or remove
```suggestion
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+On one breadboard there is a white jumper wire sending 5 volts to the potentiometer. The orange cable is connected to GND(Ground) and the gray wire is our input to A0. A0 will read the variable voltage coming in from pin13 after being adjusted through the potentiometer.
+On the right breadboard we have a simple LED circuit comprised of a red output wire from pin11, a resistor and the LED finally connected to GND via the green wire.
+
+![Full circuit of the setup.](assets/circuit.png)
```
@jhansson-ard We will need to replace this with an schematic, I'll send the draft

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+![Full circuit of the setup.](assets/circuit.png)
+
+***Make sure to connect the longer leg of the LED to the resistor and the shorter to GND.***
```
```suggestion
***Make sure to connect the longer leg of the LED (anode) to the resistor and the shorter (cathode) to GND.***
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+## Programming the Board
+
+**1.** Set your output to 1 in the previously mentioned way using `writeDigitalPin(a, 'D13', 1);`. As mentioned earlier, 1 means ON or HIGH and sends a 5V current through pin13. Check to see if A0 is actually reading an adjustable voltage using `readVoltage(a, 'A0');` in a loop and then use `display(Volts)` to print the value of `Volts`.
```
```suggestion
**1.** Set your output to 1 in the previously mentioned way using `writeDigitalPin(a, 'D13', 1);`. As mentioned earlier, 1 means ON or HIGH and sets the digital pin 13 **D13** to 5V allowing the current flow towards the LED. Check to see if the analog pin 0 **A0** is actually reading an adjustable voltage using `readVoltage(a, 'A0');` in a loop and then use `display(Volts)` to print the value of `Volts` on the "Command Window".
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
````diff
@@ -0,0 +1,148 @@
+end
+```
+
+The amount assigned to `Volts` should be changing as you turn the potentiometer.
````
```suggestion
The amount assigned to `Volts` should be changing as you rotate the potentiometer's knob.
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+![Voltage display](assets/Volts.PNG)
+
+**2.** Set the value of `Volts` as the voltage to be sent through pin 11 using the command: `writePWMVoltage()`. This command is different from `writeDigitalPin()` as its voltage is adjustable between 1 and 5 volts. Whereas the latter is either ON(1), being **5V** or OFF(0), **0V**.
```
```suggestion
**2.** Set the value of `Volts` as the voltage to be set on digital pin 11 **D11** using the command: `writePWMVoltage()`. This command is adjustable between 0 and 5 volts. It will convert the value to a PWM Signal on the digital pin.
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+### Troubleshoot
+
+If the code is not working, there are some common issues we can troubleshoot:
```
```suggestion
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+If the code is not working, there are some common issues we can troubleshoot:
+
+- Make sure to clear previous Arduino objects linked to the device. You can do this by running the command `clear a` where `a` is the Arduino object.
```
```suggestion
- If the code get frozen or it seems not working try to clear previous Arduino objects linked to the device. You can do this by running the command `clear <object>`, for example `clear a` will clear the Arduino object.
```

marqdevx commented on content/hardware/02.hero/boards/uno-rev3/tutorials/PWM-using-MATLAB-Tutorial/PWM-using-MATLAB-Tutorial.md:
```diff
@@ -0,0 +1,148 @@
+
+- You can also use the command line arguments to upload the MATLAB® server to the Arduino® UNO via the `arduino()` command. See more information on using this function in the [MathWorks® documentation](https://www.mathworks.com/help/supportpkg/arduinoio/ref/arduino.html).
+- Try writing a MATLAB® code that slowly increases and then decreases the blink speed instead of adjusting the brightness.
+- Use different sensors and parts instead of the potentiometer and LED (maybe a light sensor and a motor?)
```
A motor is not a sensor
```suggestion
- Use different sensors and parts instead of the potentiometer and LED, for example a light sensor.
```

MessiahJa replied:
Shall I remove "parts" also?

marqdevx reviewed (changes requested):
Please make sure to align the Matlab code blocks, in some of the lines you use the semicolon and in others you dont, please check it out!

marqdevx commented:
Also let me know a shorter name for the folder, it will be the link for the tutorial
I was thinking on `matlab-dimmer` the link would be `/tutorials/uno-rev3/matlab-dimmer`

marqdevx commented:
I will also change the history of commits and ammend some of them, 75 commits for one tutorial is massive, I will make another branch so you can approve my changes before applying them to this one 👌

per1234 commented:
>75 commits for one tutorial is massive

@marqdevx you can use the "**Squash and merge**" option when merging a pull request if you want to merge it as a single commit:

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits

That approach is useful for pull requests that only contain a lot of sub-atomic "fix up" commits in addition to a single primary commit, as is likely the case here.

In other cases, a pull request may contain multiple intentionally separated atomic commits. In that case, "**Squash and merge**" should not be used and one of the other merge options ("**Create a merge commit**" or "**Rebase and merge**") will be more appropriate.

marqdevx commented:
@per1234 You are right, that's a much better option 👌 
Thank you! 😄

MessiahJa commented:
> Please make sure to align the Matlab code blocks, in some of the lines you use the semicolon and in others you dont, please check it out!

Hi
I made sure the code blocks match the ones in the images. The semicolons are placed wherever they are necessary but can also be put at the end of each line, although they'd be unnecessary.

MessiahJa commented:
> > 75 commits for one tutorial is massive
> 
> @marqdevx you can use the "**Squash and merge**" option when merging a pull request if you want to merge it as a single commit:
> 
> https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits
> 
> That approach is useful for pull requests that only contain a lot of sub-atomic "fix up" commits in addition to a single primary commit, as is likely the case here.
> 
> In other cases, a pull request may contain multiple intentionally separated atomic commits. In that case, "**Squash and merge**" should not be used and one of the other merge options ("**Create a merge commit**" or "**Rebase and merge**") will be more appropriate.

Can I do this myself on GitHub desktop?

marqdevx commented:
> > Please make sure to align the Matlab code blocks, in some of the lines you use the semicolon and in others you dont, please check it out!
> 
> Hi I made sure the code blocks match the ones in the images. The semicolons are placed wherever they are necessary but can also be put at the end of each line, although they'd be unnecessary. @marqdevx

Im not gonna try the matlab code, but if it is working it's fine, i forgot about the semicolons on matlab 😄

marqdevx commented:
> > > 75 commits for one tutorial is massive
> > 
> > 
> > @marqdevx you can use the "**Squash and merge**" option when merging a pull request if you want to merge it as a single commit:
> > https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-pull-request-commits
> > That approach is useful for pull requests that only contain a lot of sub-atomic "fix up" commits in addition to a single primary commit, as is likely the case here.
> > In other cases, a pull request may contain multiple intentionally separated atomic commits. In that case, "**Squash and merge**" should not be used and one of the other merge options ("**Create a merge commit**" or "**Rebase and merge**") will be more appropriate.
> 
> Can I do this myself on GitHub desktop?

Only people with permissions can merge the PRs, so no worries :)

marqdevx commented:
First of all, im gonna rebase your branch, to fix some linter issues.
I think the content is ready to be public 🤔

marqdevx commented:
@MessiahJa Could you check this preview please?
https://docscontent-marqdevxcommunitymatlabpwmunop.gtsb.io/tutorials/uno-rev3/PWM-using-MATLAB-Tutorial

marqdevx commented:
I've changed the tutorial folder (the final link) to 
https://docs.arduino.cc/tutorials/uno-rev3/ `matlab-pwm-blink`

marqdevx reviewed (approved):
@MessiahJa content seems ready, im waiting for your confirmation 😄

MessiahJa commented:
> https://docscontent-marqdevxcommunitymatlabpwmunop.gtsb.io/tutorials/uno-rev3/PWM-using-MATLAB-Tutorial

I'm having issues opening this page

> @MessiahJa content seems ready, im waiting for your confirmation 😄

Cant wait, lets go.

marqdevx commented:
What does it say when you access the preview

Then lets merge 💪

marqdevx commented:
@MessiahJa It is public https://docs.arduino.cc/tutorials/uno-rev3/matlab-pwm-blink

It appears on both Uno R3 and UNO R3 SMD

Thanks for the contribution!! 😄

MessiahJa commented:
> What does it say when you access the preview

It worked. Thank you very much for your help @jhansson-ard, @per1234 & @marqdevx
